### ISSUE TITLE
"Source" as a part of the active process name

### ISSUE THREAD

**Issue by PavelBal** [Open-Systems-Pharmacology/MoBi#446]
When creating an actrive process (reaction or transport) in PK-Sim, the user has to enter the "Source".

When exporting a model to MoBi, the "Source" becomes part of the reaction or active transport name. I see a big problem here, because the names become very long and confusing - e.g. if the user enters the name of the publication as source, or the DOI, or....

In case of active transporters, the name of this process is not visible except for the simulation!

![image](https://user-images.githubusercontent.com/25903040/74532551-c4152a00-4f2f-11ea-951f-b29845d8fe25.png)

Following, there is no way to change this name (or even know it before creation of the simulation).

Is there a need for incorporating the source into the name? I remember that it was not there before, which also destroyed some models and/or R scripts after the change...

**Comment by msevestre** [comment]
>Is there a need for incorporating the source into the name? 
yes

At the end of the day, the name needs to be unique. 
Name of molecule is not enough as you can have more than one process defined per molecule
So name of molecule+source is the only one unique in PKSim

> remember that it was not there before

This has been like that for as long as I can remember. How was it named before according to you?

**Comment by StephanSchaller** [comment]
It is certainly not an issue that it exists. I believe what is confusing is the name "Data Source". In the end it is just a name (string) you need to define, so why not call it that ("Name" or "Tag" or "Name tag"?
Also, the option not to give any name for the first instance should be possible.

**Comment by msevestre** [comment]
It's not the name, so name does not make sence.
Why is Data source more confusing than name tag?  (especially because people have been using it for years).

And why would we have a specific rule for the first instance?  I disagree with this.

**Comment by StephanSchaller** [comment]
Yes, and it has been confusing for years ;-) … because it is not necessarily related to a "data source". You might just want to have several instances for different species.

> And why would we have a specific rule for the first instance? I disagree with this.

Technically it is not a different rule. The name would be an empty string "" which you just can't define a second time (as for other situations when you want to name similar things).

**Comment by msevestre** [comment]
>Yes, and it has been confusing for years ;-)

Say who? I never saw anyone complaining about this in training. Ever. And it's documented
https://docs.open-systems-pharmacology.org/working-with-pk-sim/pk-sim-documentation/pk-sim-compounds-definition-and-work-flow#adme-properties

Anyways

**Comment by StephanSchaller** [comment]
Well, I don't think that no-one complaining is a valid argument (and Pavel and Myself are complaining ;-)). After all, we are changing a lot of things which no-one ever complained about.

And within our workshops we did have to explain this from time to time.

And as I said: the documentation ("Enter a name for the data source (e.g. in vitro assay, literature, laboratory results).") is not correct as you might want to give it a name for other reasons than a "data source".

**Comment by PavelBal** [comment]
> Say who? I never saw anyone complaining about this in training. Ever. And it's documented

We are talking about how it is represented in MoBi. No one really cares how it is shown in PK-Sim, because you never have to deal with reactions/transports names there.

OK, here an example:

I, a user, add a linear metabolization process for a compound. What do I enter for "Data source"? Well, the paper I have the information from - "Takebayashi_2017_Effect of Sodium Glucose Cotransporter 2 Inhibitors With Low SGLT2/SGLT1 Selectivity on Circulating Glucagon-Like Peptide 1 Levels in Type 2 Diabetes Mellitus". (And what if I do not have a "source", what if I just made this process up? Do I enter "Head" as a source? ;))

However, I am not sure whether I should model this metabolization as a linear or saturable process - so I try to create a Michaelis-Menten process for the same metabolization. Once again, I enter "Takebayashi_2017_Effect of Sodium Glucose Cotransporter 2 Inhibitors With Low SGLT2/SGLT1 Selectivity on Circulating Glucagon-Like Peptide 1 Levels in Type 2 Diabetes Mellitus" as "Data source" - but I cannot because "'Takebayashi_2017_Effect of Sodium Glucose Cotransporter 2 Inhibitors With Low SGLT2/SGLT1 Selectivity on Circulating Glucagon-Like Peptide 1 Levels in Type 2 Diabetes Mellitus' already exists in Compound" (btw. why "Compound"? Either substitute with compound's name or do not capitalize...). 

My reaction as a user "Well, but it **is** the data source for this process!!!!" So maybe it is a name of the process? In this case I can understand that no two processes with identical names can exist, but

> It's not the name, so name does not make sence.

;)

OK, I am annoyed but I enter the DOI of the paper instead of its full name for the second process. I also define some active transports and export the simulation to MoBi...

And see such beauties like 

![image](https://user-images.githubusercontent.com/25903040/74646124-860b4680-5179-11ea-8926-6198807971a9.png)

![image](https://user-images.githubusercontent.com/25903040/74646162-97545300-5179-11ea-8047-0d81c659b646.png)


Changing e.g. the `Intrinsic clearance` parameter in R would look like:

`param <- getParameter("C-peptide-CYP3A4-Takebayashi_2017_Effect of Sodium Glucose Cotransporter 2 Inhibitors With Low SGLT2/SGLT1 Selectivity on Circulating Glucagon-Like Peptide 1 Levels in Type 2 Diabetes Mellitus|Intrinsic clearance", sim)`

I hope that we can agree that no one wants to see such things in the code.

And only if I knew that the entry in the `Data source` field will define the **name** of the reaction, I would have named it e.g. "Linear" and "MM"! `Data source` is more like `Value Origin`, this is where I would expect it.

In case of active transports, it gets even crazier, because I cannot change this name! I can re-name the reaction, but no way to do it for the transportrs - actually the user even do not know that there is this name!

![image](https://user-images.githubusercontent.com/25903040/74646593-76403200-517a-11ea-818c-063606f7ffdf.png)

Let's say I want to use the "Km"-value defined in an active transport process somewhere else, e.g. in a reaction. How would I construct the path to this parameter? Well, according to the `Possible Reference Objects`-list:
![image](https://user-images.githubusercontent.com/25903040/74646854-e51d8b00-517a-11ea-959a-c1e6ab71ddd3.png)


it would be `Glucose|GLUT1|Km_BBB` - which would fail, because if I drag-and-drop the entry (and remember, we cannot always use drag-and-drop, e.g. when editing an existing formula), the path is:

`Glucose|GLUT1-fuhrmann2010|Km_BBB`

Where does this `GLUT1-fuhrmann2010` come from??? And actually I just realized there is a typo and the year is 2011 and not 2010, but.... I cannot edit this name.

And now it gets even funnier - if I have **two** molecules-BBs where glucose is transported by GLUT1, but the **data source** is different, the path generated by drag-and-drop is only valid for one the BBs, and I have no idea to which one, and I also have no idea how this process is called in the other BB, and eventually I will not be able to create a simulation with the second BB ever.

I hope I could convince @msevestre that the current state is **not** the most convenient one. And only because no one ever complained does not mean that no one ever had an issue with that.

Now to the constructive part:

> Name of molecule is not enough as you can have more than one process defined per molecule

Totally agree. That's why I propose to add a field `Name` with a proper description and make it mandatory. The field `Data source` should remain but be optional and be transfered to the `Value of Origin` field.

> This has been like that for as long as I can remember. How was it named before according to you?

Right, reaction's name were like this. But not molecules. E.g., in the "old" version of the workshop, the compound was called `ADC-TMEFF2 Complex`, while now it is `ADC-TMEFF2-Boswell et al. Complex` when exported from PK-Sim.

```
Say who? I never saw anyone complaining about this in training. Ever. And it's documented
https://docs.open-systems-pharmacology.org/working-with-pk-sim/pk-sim-documentation/pk-sim-compounds-definition-and-work-flow#adme-properties
```

The documentation says "Enter a name for the data source (e.g. in vitro assay, literature, laboratory results).", which is non-informative at all regarding the naming of the process in MoBi (and this is what this topic is about), or am I missing something? In fact, the documentation is wrong, because it states:

- Right click on the biological process you want to add to (e.g. Metabolizing Enzymes in the Metabolism branch, Renal Clearances in the Transport & Excretion branch, …).
- Click on the Add … command (e.g. Add Metabolizing Enzyme …).
- **Enter a name** for the biological process you want to add.

There is no `name` to enter:

![image](https://user-images.githubusercontent.com/25903040/74648371-d2f11c00-517d-11ea-9813-d2941477f6b5.png)

**Comment by PavelBal** [comment]
Soooo....? :D

**Comment by msevestre** [comment]
Make a PR for the documentation. That's the easiest at the moment. 

As for the length of the source, if a user you keeps putting 225 long name for the source after seeing that it might be a problem, then our user is not that clever.
